### ISSUE TITLE
guild challenges don't always show up.

### ISSUE THREAD

**Issue by PeaceCaleb** [jeffhill/Dota2#75]
There are many times with guild parties with 3 or more people, the guild challenge does not show up when in the match and what players do in the match don't apply to the challenge. Attempts to recreate the party sometimes temporarily fix this but not always.

**Comment by somethingxful49** [comment]
In these cases you'll need to provide matchid's and possibly identify who is in the guild with you in party. Never heard of this bug but upvoting for visibility!

**Comment by dguillot** [comment]
Match ID: 6598918617

See screenshot for details.

5-man party in the same Guild.
Guild challenge never popped up nor counted towards it.


![3453453454](https://user-images.githubusercontent.com/6075684/171750266-7815a963-627c-4f97-86bb-4e8dac6e2fdc.jpg)

**Comment by felberjl** [comment]
This has been happening constantly in my guild for well over a year and possibly the entire time.  I'd say almost half of the valid games don't include the guild challenge.  It's really frustrating when that's the only reason we're playing a game sometimes.

This occurs with 3, 4, or 5 person parties and I haven't noticed any pattern between them.  All guild members have been part of the guild for months to years.  We've tried waiting several minutes between games and reforming the party as was mentioned above.  Anecdotally, the waiting seemed to make it somewhat more reliable, but even then plenty of games start without the challenge.

I'll try to find an example next time I play.

**Comment by dguillot** [comment]
Same for me.

My theories were:

- Adding a coach
- Removing a coach
- Losing someone in your party while queuing
- Someone joining your party while you are in-game with the other members
- Changing party leader.

I gave up when one game it was working, the other wasn't working without ANY changes whatsoever.

I've never reported it because it is so obvious that the devs at Valve would have figured this out already. Seems like they never did. Do they have QAs?

**Comment by dtseiler** [comment]
God yes please fix this

**Comment by dguillot** [comment]
Match ID: 6601850796

See screenshot for details.

5-man party in the same Guild.
Guild challenge never popped up nor counted towards it.
![1 copy](https://user-images.githubusercontent.com/6075684/172023225-1bbfc7b8-642c-4867-95ea-386e839fe343.png)
![2](https://user-images.githubusercontent.com/6075684/172023227-08b98ed9-e7b8-494e-b5f3-10f04c51e7af.png)

**Comment by raduiordache** [comment]
I have one as well:

Match ID: 6603624119

3 of the 5 in our team were from the same guild. The challenge didn't count, although the game before it did. From my experience, the guild challenge doesn't activate every 3-4 games.

**Comment by dtseiler** [comment]
Match ID: 6613581371

4 of our 5 were in my guild. Previous matches we played today had the challege, this is the first one today that I noticed did not.

**Comment by felberjl** [comment]
Match ID: 6615391931

4 of 5 in guild.

**Comment by MrVdota** [comment]
I've had this bug as well. I've noticed that if a player in the guild disconnects during the draft phase, then this may also prevent the guild challenge from showing up. 
Match ID: 6639309780 (3 in guild. Lina, Axe, Sniper).

**Comment by dtseiler** [comment]
I haven't noticed any link with disconnects. I've had plenty of games with no disconnects and once we enter the game we can see if the guild challenge is in effect. My guildmates and I have been scrambling trying to find any pattern or narrow down causes and so far nothing has stuck.

**Comment by mouyipen** [comment]
that's cool

**Comment by MrVdota** [comment]
I wonder if this issue is related to how guild challenges are saved/updated on the server side. To test this hypothesis, we would need to make a new guild with only 3-members that ONLY play guild challenges together once every day. This will give the server ample time to update and so that no random guild members generate progress towards the weekly quest. Then, if the guild challenge ever spontaneously breaks, then we would know that something else is going on.

**Comment by jovicacar** [comment]
Yes, pls fix this. This bug is now over 2 years old. Come on...

**Comment by dtseiler** [comment]
BUMP

**Comment by Orbys** [comment]
This post was made over a year ago, but I can assure you it began happening when Guilds were first introduced back in October of 2020.  Its still happening today, and it feels worse than ever.  My guild mates and I are getting so burnt out on this bug ruining it for us.  

We constantly have stacks of 3 or more guild members in our parties and the weekly quest doesn't trigger 3/4ths of the time.

**Comment by LokeLok** [comment]
I haven't seen anyone pointing out, that this is only an issue in Turbo. It works fine otherwise.

**Comment by jovicacar** [comment]
> I haven't seen anyone pointing out, that this is only an issue in Turbo. It works fine otherwise.

No, I rarely play turbo. This is an issue in Normal and Ranked games. So frustrating...

**Comment by hgajshb** [comment]
In https://store.steampowered.com/news/app/570/view/3640648066072340345 they actually want more bug reports.

**Comment by lincolnpjames** [comment]
> 

Affects both Ranked/Turbo games

Just some more match id's

7285978184
7285953892

**Comment by Stresing** [comment]
Many people have this problem, but not many people complain about it. it is sad.
Fix this, the bug will soon celebrate its anniversary

**Comment by felberjl** [comment]
This has still never been corrected.  We only get the guild challenge for the team to show up about half the time.  No disconnects or party changes necessarily made at any time between working and non-working games.  Recent example of not working: 8592216945

**Comment by asinkrit84-web** [comment]
This bug is now over 6 years old.
